packages/dds/tree/src/feature-libraries/defaultFieldKinds.ts

```diff
@@ -189,18 +194,144 @@ export const counter: FieldKind = new FieldKind(
+    changes?: NodeChangeset;
+}
+
+const valueRebaser: FieldChangeRebaser<ValueChangeset> = {
+    compose: (changes: ValueChangeset[], composeChildren: NodeChangeComposer): ValueChangeset => {
+        if (changes.length === 0) {
+            return {};
+        }
+        let newValue: JsonableTree | undefined;
+        const childChanges: NodeChangeset[] = [];
+        for (const change of changes) {
+            if (change.value !== undefined) {
+                newValue = change.value;
+
+                // The previous changes applied to a different value, so we discard them.
+                // TODO: Consider if we should represent muted changes
+                childChanges.length = 0;
+            }
+
+            if (change.changes !== undefined) {
+                childChanges.push(change.changes);
+            }
+        }
+
+        const composed: ValueChangeset = {};
+        if (newValue !== undefined) {
+            composed.value = newValue;
+        }
+
+        if (childChanges.length > 0) {
+            composed.changes = composeChildren(childChanges);
+        }
+
+        return composed;
+    },
+
+    invert: (change: ValueChangeset, invertChild: NodeChangeInverter): ValueChangeset => {
+        // TODO: Handle the value inverse
+        const inverse: ValueChangeset = {};
+        if (change.changes !== undefined) {
+            inverse.changes = invertChild(change.changes);
+        }
+        return inverse;
+    },
+
+    rebase: (change: ValueChangeset, over: ValueChangeset, rebaseChild: NodeChangeRebaser): ValueChangeset => {
+        if (change.changes === undefined || over.changes === undefined) {
+            return change;
+        }
+        return { ...change, changes: rebaseChild(change.changes, over.changes) };
+    },
+};
+
+interface EncodedValueChangeset {
+    value?: JsonableTree;
+    changes?: JsonCompatibleReadOnly;
+}
+
+const valueFieldEncoder: FieldChangeEncoder<ValueChangeset> = {
+    encodeForJson: (formatVersion: number, change: ValueChangeset, encodeChild: NodeChangeEncoder) => {
+        const encoded: EncodedValueChangeset & JsonCompatibleReadOnly = {};
+        if (change.value !== undefined) {
+            encoded.value = change.value;
+        }
+
+        if (change.changes !== undefined) {
+            encoded.changes = encodeChild(change.changes);
+        }
+
+        return encoded;
+    },
+
+    decodeJson: (formatVersion: number, change: JsonCompatibleReadOnly, decodeChild: NodeChangeDecoder) => {
+        const encoded = change as EncodedValueChangeset;
+        const decoded: ValueChangeset = {};
+        if (encoded.value !== undefined) {
+            decoded.value = encoded.value;
+        }
+
+        if (encoded.changes !== undefined) {
+            decoded.changes = decodeChild(encoded.changes);
+        }
+
+        return decoded;
+    },
+};
+
+export interface ValueFieldEditor extends FieldEditor<ValueChangeset> {
+    setValue(newValue: JsonableTree): ValueChangeset;
```

Comment:
I think "setValue" is too easily confused with `ModularEditBuilder.setValue`. To be fair, the confusion originates in having both values on nodes and value-kind fields, but since we're not changing that for now, I think a different same would be good.
```suggestion
    setTree(newTree: JsonableTree): ValueChangeset;
```
```suggestion
    set(newTree: JsonableTree): ValueChangeset;
```

Reply:
What do you think about calling this `replace`?

Reply:
I do think that the editing semantics are best described in english as a replacement, the only issue with replace is that in english, "replace foo" means "foo" is the old thing not the new thing. We could go with `replaceWith` but it's rather clunky for something so common and basic.
Out of all these, I have a slight preference for `set` but I'm happy with `replace` too. We can get feedback from more people later.